mobile_rewards/src/pending_txn.rs

```diff
@@ -148,17 +162,16 @@ impl PendingTxn {
             .bind(hash)
             .bind(timestamp.into())
             .execute(executor)
-            .map_ok(|_| ())
-            .map_err(Error::from)
-            .await
+            .await?;
+        Ok(())
```

Comment:
is there a reason we're wrapping the previous function/method chain in an `Ok()` and here we're just executing the chain and then returning an empty `Ok(())`? neither sql statement is a "get" request so i could see an argument both for and against returning the record being "upserted" in either case so these feel like they could be aligned

Reply:
While I think you can consider nil a valid return value from an expression and therefore it should go in the Ok, I think that the returning of nil usually signals state effects and therefore should be written sequentially